ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -54,6 +54,9 @@ def __init__(self, filter: Filter, team: Team) -> None:
         self._filter = filter
         self._team = team
 
+        if self._filter.funnel_step is None:
+            self._filter = self._filter.with_data({"funnel_step": len(self._filter.entities)})
```

Comment:
It's not obvious what this is meant to do, does this just set the `funnel_step` attribute on `_filter`? What's the difference to `self._filter.funnel_step = len(self._filter.entities)`? aside from mutation.

Reply:
Good question! The problem arises when: 

1. The frontend wouldn't send `funnel_step` in the request, because it's not important to correlation analysis (it's a backend implementation detail)
2. Which means, the funnel_cte validation fails, because funnel_step isn't set.

The above ensures that the funnel_step is set on the filter post request coming in, so the cte has no issues, and the API remains clean

Reply:
Given that this is specific to the `cte`, could be move to the generating function?

Reply:
i.e. `get_funnel_persons_cte`

Reply:
I could do that.

The only qualm: I feel icky about mutating the filter at all. And if I have to do it, I want to do it at the topmost level, so I don't have to keep in mind how functions below modify the filter. Thoughts?